browser/net/brave_service_key_network_delegate_helper.cc

```diff
@@ -0,0 +1,32 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/net/brave_service_key_network_delegate_helper.h"
+
+#include <memory>
+
+#include "brave/common/network_constants.h"
+#include "extensions/common/url_pattern.h"
+#include "net/base/net_errors.h"
+#include "net/http/http_request_headers.h"
+
+namespace brave {
+
+int OnBeforeStartTransaction_BraveServiceKey(
+    net::HttpRequestHeaders* headers,
+    const ResponseCallback& next_callback,
+    std::shared_ptr<BraveRequestInfo> ctx) {
+  static URLPattern brave_proxy_pattern(URLPattern::SCHEME_HTTPS,
```

Comment:
frankly speaking static objects like these are prohibited by CC, but why don't we just use GURL::DomainIs?

Reply:
resolved